test/cpp/cpp_test_util.cpp

```diff
@@ -78,8 +78,15 @@ at::Tensor ToCpuTensor(const at::Tensor& tensor) {
 }
 
 torch::Tensor CopyToDevice(const torch::Tensor& tensor,
-                           const torch::Device& device) {
-  return tensor.to(device, /*non_blocking=*/false, /*copy=*/true);
+                           const torch::Device& device,
+                           bool requires_grad) {
+  auto output = tensor.to(device, /*non_blocking=*/false, /*copy=*/true);
+  if (requires_grad) {
+    // The .detach().requires_grad_(true) is to make sure that the new tensor is a leaf in the autograd graph.
+    return output.detach().set_requires_grad(true);
```

Comment:
the detach().set_requires_grad(true) is also to appease autograd. We need the xla tensor "input" for each test to be a "leaf node", from autograd's perspective.

Reply:
so... when copying tensor using `tensor.to`, will `require_grad` field also being copied over? I am guessing you need to reset `set_requires_grad` because you called `deteach` but want to double check. 😄

Reply:
didn't know until I made this PR, but yep 😛 

```
device = xm.xla_device()
a = torch.ones(2, 2, device='cpu', requires_grad=True)
b = a.to(device)
print(b.requires_grad) # printed "True"
```